qiskit_metal/analyses/core/base.py

```diff
@@ -84,6 +91,16 @@ def run_sim(self, *args, **kwargs):
         QAnalysis subclass, in case you expect them to be both inherited from the same subclass.
         """
 
+    def run_sweep(self, *args, **kwargs):
+        """User requests sweeper based on arguments from
+        Sweeper.run_sweep().
+        """
+        if not self._sweeper:
+            self._initialize_sweep()
+
```

Comment:
Ideally we will not "require" users to pass args[3]->....
That is because user might have already run analysis, which stores their input string in `c1.setup.run` (See `save_run_args()`). So the run_sweeps() [either the one here or the won in the sweeper], can check if args[3]->.... are passed and if they are not, it can go search for them in `c1.setup.run`, and give an error if it does not find them. Thoughts?

Reply:
> Ideally we will not "require" users to pass args[3]->....
> That is because user might have already run analysis, which stores their input string in `c1.setup.run` (See `save_run_args()`). So the run_sweeps() [either the one here or the won in the sweeper], can check if args[3]->.... are passed and if they are not, it can go search for them in `c1.setup.run`, and give an error if it does not find them. Thoughts?

Since the user is providing component option to sweep, the Sweeper.run_sweep needs to run analysis each time.  I can see other scenarios when checking for c1.setup.run would be useful, but not for Sweeping.